src/OAuth.js

```diff
@@ -0,0 +1,68 @@
+/* * *  *  * *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  *  * */
```

Comment:
As per out conversation, I think the client should not have the any tooling exposed for getting oAuth tokens aside from a utility used for testing. So we might want to move this function into test/util or something similar.